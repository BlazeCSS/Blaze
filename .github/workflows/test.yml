name: CI

on:
  push:
    branches:
      - master

jobs:
  build:
    name: Install, test and build
    
    runs-on: ubuntu-latest
    
    steps:
      - name: Checkout repo
        uses: actions/checkout@master

      - name: Install dependencies
        run: npm ci

      - name: Test
        run: npm run test

      - name: Build site
        run: npm run build
      
      - name: Switch to site dir
        run: cd ./packages/site

      - name: Deploy site
        uses: w9jds/firebase-action@master
        with:
          args: deploy --only hosting
        env:
          FIREBASE_TOKEN: ${{ secrets.FIREBASE_TOKEN }}
          PROJECT_ID: "default"
