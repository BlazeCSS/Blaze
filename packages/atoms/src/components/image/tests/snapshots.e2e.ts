import { snapIt } from '../../../test';

const component = 'image';

describe(component, () => {
  const snap = snapIt(component);

  describe('renders', () => {
    snap(
      'a standard image with src and alt attributes',
      '<blaze-image src="https://www.blazeui.com/assets/img/blaze.png" alt="blaze logo"></blaze-image>'
    );
    snap('fall back to a random placeholder image', '<blaze-image></blaze-image>');
    snap('fall back to a random placeholder image with a filter', '<blaze-image filter="space"></blaze-image>');
    snap('width and height', '<blaze-image width="1024" height="768"></blaze-image>');
    snap('with specific photo id', '<blaze-image photo="test-photo-id"></blaze-image>');
    snap(
      'specific photo id and width and height',
      '<blaze-image photo="test-photo-id" width="1024" height="768"></blaze-image>'
    );
    snap('specific user', '<blaze-image user="test-user"></blaze-image>');
    snap(
      'specific user with width and height',
      '<blaze-image user="test-user" width="1024" height="768"></blaze-image>'
    );
    snap(
      'specific user with width and height and an additional filter',
      '<blaze-image user="test-user" width="1024" height="768" filter="plane"></blaze-image>'
    );
    snap(
      "from the list of a specific user's like photos with additional attributes",
      '<blaze-image user="test-user" width="1024" height="768" filter="mountains" likes></blaze-image>'
    );
  });
});
