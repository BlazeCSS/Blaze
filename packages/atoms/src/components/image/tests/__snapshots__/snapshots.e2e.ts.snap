// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`image renders a standard image with src and alt attributes 1`] = `"<blaze-image src=\\"https://www.blazeui.com/assets/img/blaze.png\\" alt=\\"blaze logo\\" class=\\"hydrated\\"><img class=\\"o-image\\" alt=\\"blaze logo\\" src=\\"https://www.blazeui.com/assets/img/blaze.png\\"></blaze-image>"`;

exports[`image renders fall back to a random placeholder image 1`] = `"<blaze-image class=\\"hydrated\\"><img class=\\"o-image\\" src=\\"https://source.unsplash.com/random/\\"></blaze-image>"`;

exports[`image renders fall back to a random placeholder image with a filter 1`] = `"<blaze-image filter=\\"space\\" class=\\"hydrated\\"><img class=\\"o-image\\" src=\\"https://source.unsplash.com/random/?space\\"></blaze-image>"`;

exports[`image renders from the list of a specific user's like photos with additional attributes 1`] = `"<blaze-image user=\\"test-user\\" width=\\"1024\\" height=\\"768\\" filter=\\"mountains\\" likes=\\"\\" class=\\"hydrated\\"><img class=\\"o-image\\" src=\\"https://source.unsplash.com/user/test-user/likes/1024x768/?mountains\\"></blaze-image>"`;

exports[`image renders specific photo id and width and height 1`] = `"<blaze-image photo=\\"test-photo-id\\" width=\\"1024\\" height=\\"768\\" class=\\"hydrated\\"><img class=\\"o-image\\" src=\\"https://source.unsplash.com/test-photo-id/1024x768/\\"></blaze-image>"`;

exports[`image renders specific user 1`] = `"<blaze-image user=\\"test-user\\" class=\\"hydrated\\"><img class=\\"o-image\\" src=\\"https://source.unsplash.com/user/test-user/\\"></blaze-image>"`;

exports[`image renders specific user with width and height 1`] = `"<blaze-image user=\\"test-user\\" width=\\"1024\\" height=\\"768\\" class=\\"hydrated\\"><img class=\\"o-image\\" src=\\"https://source.unsplash.com/user/test-user/1024x768/\\"></blaze-image>"`;

exports[`image renders specific user with width and height and an additional filter 1`] = `"<blaze-image user=\\"test-user\\" width=\\"1024\\" height=\\"768\\" filter=\\"plane\\" class=\\"hydrated\\"><img class=\\"o-image\\" src=\\"https://source.unsplash.com/user/test-user/1024x768/?plane\\"></blaze-image>"`;

exports[`image renders width and height 1`] = `"<blaze-image width=\\"1024\\" height=\\"768\\" class=\\"hydrated\\"><img class=\\"o-image\\" src=\\"https://source.unsplash.com/random/1024x768/\\"></blaze-image>"`;

exports[`image renders with specific photo id 1`] = `"<blaze-image photo=\\"test-photo-id\\" class=\\"hydrated\\"><img class=\\"o-image\\" src=\\"https://source.unsplash.com/test-photo-id/\\"></blaze-image>"`;
